ambry-store/src/test/java/com.github.ambry.store/IndexTest.java

```diff
@@ -52,6 +52,15 @@
   private final boolean isLogSegmented;
   private final File tempDir;
   private final CuratedLogIndexState state;
+  private static final StoreKeyFactory STORE_KEY_FACTORY;
+
+  static {
+    try {
+      STORE_KEY_FACTORY = Utils.getObj("com.github.ambry.store.MockIdFactory");
```

Comment:
you can use the `STORE_KEY_FACTORY` in `CuratedLogIndexState` (make it package private if required)